### ISSUE TITLE
unconverted data remains: Z

### ISSUE THREAD

**Issue by grahampugh** [grahampugh/jamf-upload#53]
Sometimes we get this error which makes a recipe fail. I suspect it's something to do with the way we are measuring expiry of tokens.

**Comment by mpanighetti** [comment]
Saw this happening today on every JamfUploader recipe run during the JamfCategoryUploader step. Issue went away after we deleted `/private/tmp/jamf_upload` and `/private/tmp/jamf_upload_token` based on troubleshooting guidance given in MacAdmins Slack.

Traceback if it helps:

```
Traceback (most recent call last):
  File "/Library/AutoPkg/autopkglib/__init__.py", line 840, in process
    self.env = processor.process()
  File "/Library/AutoPkg/autopkglib/__init__.py", line 626, in process
    self.main()
  File "/path/to/RecipeRepos/com.github.autopkg.grahampugh-recipes/JamfUploaderProcessors/JamfCategoryUploader.py", line 119, in main
    token = self.handle_uapi_auth(self.jamf_url, self.jamf_user, self.jamf_password)
  File "/path/to/RecipeRepos/com.github.autopkg.grahampugh-recipes/JamfUploaderProcessors/JamfUploaderLib/JamfUploaderBase.py", line 225, in handle_uapi_auth
    token = self.check_api_token(url)
  File "/path/to/RecipeRepos/com.github.autopkg.grahampugh-recipes/JamfUploaderProcessors/JamfUploaderLib/JamfUploaderBase.py", line 156, in check_api_token
    data["expires"].split(".")[0], "%Y-%m-%dT%H:%M:%S"
  File "/Library/AutoPkg/Python3/Python.framework/Versions/Current/lib/python3.7/_strptime.py", line 577, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/Library/AutoPkg/Python3/Python.framework/Versions/Current/lib/python3.7/_strptime.py", line 362, in _strptime
    data_string[found.end():])
ValueError: unconverted data remains: Z
  File "/Library/AutoPkg/autopkglib/__init__.py", line 840, in process
    self.env = processor.process()
unconverted data remains: Z
Failed.
```

**Comment by grahampugh** [comment]
I have pushed a change to `JamfUploaderBase` to attempt to handle this better. It now incorporates the `Z` in the time import step, which python's `datetime` seems to accept, and I have added an exception statement that, if it can't recognise the time format, will just not return the token and a new one will be obtained. So recipes should not fail any more either way. But, let me know if you still see failures - I will also be monitoring this on our own runs of course.
